core/src/test/kotlin/io/specmatic/core/pattern/AnyOfPatternTest.kt

```diff
@@ -0,0 +1,273 @@
+        
+        val result = biggerPattern.encompasses(smallerPattern, Resolver(), Resolver())
+        assertThat(result).isInstanceOf(Result.Success::class.java)
+    }
+
+    @Test
+    fun `should not encompass incompatible patterns`() {
+        val pattern = AnyOfPattern(listOf(NumberPattern(), StringPattern()), extensions = emptyMap())
+        val incompatiblePattern = BooleanPattern()
+        
+        val result = pattern.encompasses(incompatiblePattern, Resolver(), Resolver())
+        assertThat(result).isInstanceOf(Result.Failure::class.java)
+    }
+
+    @Test
+    fun `should handle JSON object patterns`() {
+        val objectPattern1 = JSONObjectPattern(mapOf("name" to StringPattern()), typeAlias = "(Person)")
+        val objectPattern2 = JSONObjectPattern(mapOf("id" to NumberPattern()), typeAlias = "(Id)")
+        val pattern = AnyOfPattern(listOf(objectPattern1, objectPattern2), extensions = emptyMap())
+        
+        val personValue = JSONObjectValue(mapOf("name" to StringValue("John")))
+        val idValue = JSONObjectValue(mapOf("id" to NumberValue(123)))
+        
+        assertThat(pattern.matches(personValue, Resolver())).isInstanceOf(Result.Success::class.java)
+        assertThat(pattern.matches(idValue, Resolver())).isInstanceOf(Result.Success::class.java)
+    }
+
+    @Test
+    fun `should fix value using best matching pattern`() {
+        val pattern = AnyOfPattern(listOf(NumberPattern(), StringPattern()), extensions = emptyMap())
+        val value = StringValue("hello")
+        
+        val fixedValue = pattern.fixValue(value, Resolver())
+        assertThat(fixedValue).isEqualTo(value)
+    }
+
+    @Test
+    fun `should eliminate optional keys correctly`() {
+        val objectPattern = parsedPattern("""{
+            "mandatoryKey": "(string)",
+            "optionalKey?": "(number)"
+        }""")
+        val pattern = AnyOfPattern(listOf(objectPattern), extensions = emptyMap())
+        
+        val originalValue = parsedValue("""{
+            "mandatoryKey": "hello",
+            "optionalKey": 42
+        }""")
+        
+        val expectedValue = parsedValue("""{
+            "mandatoryKey": "hello"
+        }""")
+        
+        val result = pattern.eliminateOptionalKey(originalValue, Resolver())
+        assertEquals(expectedValue, result)
+    }
+
+    @Test
+    @Tag(GENERATION)
+    fun `should generate new patterns for all available types`() {
+        AnyOfPattern(
+            listOf(
+                NumberPattern(),
+                EnumPattern(listOf(StringValue("one"), StringValue("two")))
+            ),
+            extensions = emptyMap()
+        ).newBasedOn(Row(), Resolver()).map { it.value }.toList().let { patterns ->
+            patterns.map { it.typeName } shouldContainInAnyOrder listOf("number", "\"one\"", "\"two\"")
+        }
+    }
+
+    @Test
+    @Tag(GENERATION) 
+    fun `should create new pattern based on row data`() {
```

Comment:
In this test, there should be two patterns generated from the row, one of which is a number, and one a string containing "hello".